NetworkPkg/WifiConnectionManagerDxe/WifiConnectionMgrImpl.c

```diff
@@ -1491,6 +1491,14 @@ WifiMgrOnTimerTick (
   Nic = (WIFI_MGR_DEVICE_DATA *)Context;
   NET_CHECK_SIGNATURE (Nic, WIFI_MGR_DEVICE_DATA_SIGNATURE);
 
+  if ((Nic->ConnectPendingNetwork == NULL) || (StrLen (Nic->ConnectPendingNetwork->SSId) < SSID_MIN_LEN) || \

```

Comment:
Would prefer to see NULL check of ConnectPendingNetwork in its own if and only deref in side that if statement.  Also, there are 2 conditions being checked here with a debug message that says one or the other.  It would be better to check each condition independent with a clear debug message for each condition.

Reply:
Broke this into 2 checks, valid profile and current connection status.